src/Authentication/Authentication/Helpers/StreamHelper.cs

```diff
@@ -0,0 +1,178 @@
+using System;
+using System.IO;
+using System.Management.Automation;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Graph.PowerShell.Authentication.Cmdlets;
+
+namespace Microsoft.Graph.PowerShell.Authentication.Helpers
+{
+    internal class StreamHelper
+    {
+        internal const int DefaultReadBuffer = 100000;
+
+        internal const int ChunkSize = 10000;
+
+        public static byte[] EncodeToBytes(string str, Encoding encoding)
+        {
+            if (encoding == null)
+            {
+                // just use the default encoding if one wasn't provided
+                encoding = ContentHelper.GetDefaultEncoding();
+            }
+
+            return encoding.GetBytes(str);
+        }
+
+        internal static Stream GetResponseStream(HttpResponseMessage response)
+        {
+            var responseStream = response.Content.ReadAsStreamAsync().GetAwaiter().GetResult();
+            return responseStream;
+        }
+
+        internal static bool TryGetEncoding(string characterSet, out Encoding encoding)
+        {
+            var result = false;
+            try
+            {
+                encoding = Encoding.GetEncoding(characterSet);
+                result = true;
+            }
+            catch (ArgumentException)
+            {
+                encoding = null;
+            }
+
+            return result;
+        }
+
+        internal static string DecodeStream(BufferingStreamReader responseStream, ref Encoding encoding)
+        {
+            var isDefaultEncoding = false;
+            if (encoding == null)
+            {
+                // Use the default encoding if one wasn't provided
+                encoding = ContentHelper.GetDefaultEncoding();
+                isDefaultEncoding = true;
+            }
+
+            var content = StreamToString(responseStream, encoding);
+            if (isDefaultEncoding)
+            {
+                do
+                {
+                    // check for a charset attribute on the meta element to override the default.
+
+                    var localEncoding = Encoding.UTF8;
+                    responseStream.Seek(0, SeekOrigin.Begin);
+                    content = StreamToString(responseStream, localEncoding);
+                    // report the encoding used.
+                    encoding = localEncoding;
+                } while (false);
```

Comment:
Q: What is the purpose of having a loop here? This loop seems to be guaranteed to execute once.